action.yaml

```diff
@@ -449,6 +453,35 @@ runs:
         echo ::endgroup::
       shell: bash
       id: deploy-options
+    - name: Determine if Development Mode is enabled
+      if: ${{ inputs.wake-on-deploy == 'true' && inputs.action != 'delete-deployment-preview' }}
+      run: |
+        echo ::group::Determine if Development Mode is enabled
+        DEPLOYMENT_TYPE=$(astro deployment inspect ${{steps.deployment-preview.outputs.FINAL_DEPLOYMENT_ID}} --clean-output --key configuration.deployment_type)
+        # only inspect development mode if deployment is not hybrid
+        if [[ $DEPLOYMENT_TYPE != "HYBRID" ]]; then
+          DEV_MODE=$(astro deployment inspect ${{steps.deployment-preview.outputs.FINAL_DEPLOYMENT_ID}} --clean-output --key configuration.is_development_mode)
+          echo "Deployment development mode: $DEV_MODE"
+          if [[ $DEV_MODE == "" ]]; then
+            echo "DEVELOPMENT_MODE=false" >> $GITHUB_OUTPUT
+          else
+            echo "DEVELOPMENT_MODE=$DEV_MODE" >> $GITHUB_OUTPUT
+          fi
+        else
+          echo "DEVELOPMENT_MODE=false" >> $GITHUB_OUTPUT
+        fi
+        echo ::endgroup::
+      shell: bash
+      id: development-mode
+    - name: Override to wake up the Deployment
+      if: ${{ inputs.wake-on-deploy == 'true' && inputs.action != 'delete-deployment-preview' && steps.development-mode.outputs.DEVELOPMENT_MODE == 'true' }}
+      run: |
+        echo ::group::Override to wake up the Deployment
+        astro deployment wake-up ${{steps.deployment-preview.outputs.FINAL_DEPLOYMENT_ID}} --force
+        # Give it some time to wake up
+        sleep 60
```

Comment:
At first glance this seems risky, worst case 60 seconds isn't enough and it fails to deploy, but it could also be waiting around longer than it needs to. Can we poll the deployment instead? Ideally `astro deployment wake-up` would have a `--wait` flag on it though.

Reply:
Will add polling until we have a `--wait` flag as discussed offline 👍🏼